proof/invariant-abstract/Syscall_AI.thy

```diff
@@ -100,13 +104,14 @@ lemma schedule_choose_new_thread_valid_state_cur_tcb [wp]:
                wp: set_scheduler_action_valid_state_cur_tcb switch_to_idle_thread_invs
                    guarded_switch_to_invs hoare_drop_imps)
 
-lemma schedule_invs[wp]: "\<lbrace>invs\<rbrace> Schedule_A.schedule \<lbrace>\<lambda>rv. invs\<rbrace>"
-  supply if_split[split del]
-  apply (simp add: Schedule_A.schedule_def)
-  apply (wp switch_to_thread_invs hoare_drop_imps
-            hoare_strengthen_post[OF awaken_invs] sc_and_timer_invs
-         | simp add: if_apply_def2 set_scheduler_action_def invs_def
+lemma schedule_invs[wp]:
+  "schedule \<lbrace>invs\<rbrace>"
+  apply (simp add: schedule_def)
+  apply (wpsimp wp: switch_to_thread_invs hoare_drop_imps sc_and_timer_invs
+              simp: if_apply_def2 set_scheduler_action_def
          | wpc)+
```

Comment:
tiny nitpick: this shouldn't need the `wpc` or `+` anymore

Reply:
Done, thanks